lib/ansible/modules/crypto/openssl_pkcs12.py

```diff
@@ -0,0 +1,365 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: openssl_pkcs12
+author: "Guillaume Delpierre (@gdelpierre)"
+version_added: "2.4"
+short_description: Generate OpenSSL PKCS#12 archive.
+description:
+    - This module allows one to (re-)generate PKCS#12.
+requirements:
+    - python-pyOpenSSL
+options:
+    action:
+        default: export
+        choices: ['parse', 'export']
+        description:
+            - C(export) or C(parse) a PKCS#12.
+    ca_certificates:
+        description:
+            - List of CA certificate to include.
+    certificate_path:
+        description:
+            - The path to read certificates and private keys from.  Must be in PEM format.
+    force:
+        default: False
+        type: bool
+        description:
+            - Should the file be regenerated even it it already exists.
+    friendly_name:
+        aliases: 'name'
+        description:
+            - Specifies the friendly name for the certificate and private key.
+    iter_size:
+        default: 2048
+        description:
+            - Number of times to repeat the encryption step.
+    maciter_size:
+        default: 1
+        description:
+            - Number of times to repeat the MAC step.
+    mode:
+        default: 0400
+        description:
+            - Default mode for the generated PKCS#12 file.
+    passphrase:
+        description:
+            - The PKCS#12 password.
+    path:
+        required: True
+        description:
+            - Filename to write the PKCS#12 file to.
+    privatekey_passphrase:
+        description:
+            - Passphrase source to decrypt any input private keys with.
+    privatekey_path:
+        description:
+            - File to read private key from.
+    state:
+        default: 'present'
+        choices: ['present', 'absent']
+        description:
+            - Whether the file should exist or not.
```

Comment:
All parameters exept `path` are ignored when `state` is `absent`, this could be mentioned here.